bin/describe/describe_pp.ml

```diff
@@ -19,28 +19,44 @@ let pp_with_ocamlc env ~ocamlc pp_file dump_file =
     User_error.raise [ Pp.textf "cannot find a dump file: %s" (Path.to_string dump_file) ]
 ;;
 
-let files_for_source file dialects =
+let dump_file pp_file ~ml_kind =
+  Path.set_extension
+    pp_file
+    ~ext:
+      (match (ml_kind : Ocaml.Ml_kind.t) with
+       | Intf -> ".cmi.dump"
+       | Impl -> ".cmo.dump")
+;;
+
+let files_for_source file ~dialects =
   let base, ext = Path.split_extension file in
   let dialect, kind =
     match Dune_rules.Dialect.DB.find_by_extension dialects ext with
     | None -> User_error.raise [ Pp.textf "unsupported extension: %s" ext ]
     | Some x -> x
   in
   let pp_file_base = Path.extend_basename base ~suffix:ext in
-  let pp_file =
+  let pp_files =
+    let pp_file_result_base =
+      Path.map_extension pp_file_base ~f:(fun ext -> ".pp" ^ ext)
+    in
     match Dune_rules.Dialect.ml_suffix dialect kind with
-    | None -> pp_file_base
-    | Some suffix -> Path.extend_basename pp_file_base ~suffix
-  in
-  let dump_file =
-    Path.set_extension
-      pp_file
-      ~ext:
-        (match kind with
-         | Intf -> ".cmi.dump"
-         | Impl -> ".cmo.dump")
+    | None ->
+      (* No `.ml` suffix for this dialect. this is already an `.ml` file.
+         The extension for the pped file is `.pp.ml` *)
+      [ pp_file_result_base ]
+    | Some suffix ->
+      (* If there's an `.ml` suffix for this dialect, append it to the source.
```

Comment:
I find this check to be unnecessary. Why do we need to guess if a file exists or not if we are the ones setting up the rules? Can't we just know what the source should be from `Module.t`?

Reply:
I guess so. What's the best way to get a `Module.t` from the source path that the user passes in the CLI to describe pp?